tools/Import/Importer.cs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Health.Fhir.Store.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Import
+{
+    internal static class Importer
+    {
+        private static readonly HttpClient HttpClient = new HttpClient();
+        private static readonly string ConnectionString = ConfigurationManager.AppSettings["ConnectionString"];
+        private static readonly string FhirEndpoint = ConfigurationManager.AppSettings["FhirEndpoint"];
+        private static string fhirEndpoint2 = ConfigurationManager.AppSettings["FhirEndpoint2"];
+        private static readonly string ContainerName = ConfigurationManager.AppSettings["ContainerName"];
+        private static readonly int BlobRangeSize = int.Parse(ConfigurationManager.AppSettings["BlobRangeSize"]);
+        private static readonly int ReadThreads = int.Parse(ConfigurationManager.AppSettings["ReadThreads"]);
+        private static readonly int WriteThreads = int.Parse(ConfigurationManager.AppSettings["WriteThreads"]);
+        private static readonly int NumberOfBlobsToSkip = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsToSkip"]);
+        private static readonly int NumberOfBlobsToImport = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsToImport"]);
+        private static readonly int ReportingPeriodSec = int.Parse(ConfigurationManager.AppSettings["ReportingPeriodSec"]);
+        private static readonly int MaxRetries = int.Parse(ConfigurationManager.AppSettings["MaxRetries"]);
+        private static readonly bool UseStringJsonParser = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParser"]);
+        private static readonly bool UseStringJsonParserCompare = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParserCompare"]);
+
+        private static long readers = 0L;
+        private static long resourcesRead = 0L;
+        private static Stopwatch swReads = Stopwatch.StartNew();
+
+        internal static void Run()
+        {
+            ServicePointManager.DefaultConnectionLimit = 2048;
+
+            if (string.IsNullOrEmpty(fhirEndpoint2))
+            {
+                fhirEndpoint2 = FhirEndpoint;
+            }
+
+            var globalLogPrefix = $"GlobalBlobRange=[{NumberOfBlobsToSkip + 1}-{NumberOfBlobsToImport}]";
+            Console.WriteLine($"{globalLogPrefix}: Starting...");
+            var blobContainerClient = GetContainer(ConnectionString, ContainerName);
```

Comment:
Is it better to define the blobContainerClient as a global variable than a local variable? This way, it doesn't need to be created again in the GetLinesInBlobRange method.

Reply:
The problem is that retry logic inside Azure client gets exceptions sometimes. We will need to retry that. Creating new containers is not that expensive compared to all the work we do.